File: LabourBudgetCalculator/Models/RateSheet.cs
using System;
using System.Collections.Generic; // Not strictly needed for this class definition but often included
using System.Xml.Serialization; // Required for [Serializable]

namespace TimeExpenseCalculator.Models // As per your provided DataManager.cs
{
    [Serializable] // For XML serialization
    public class RateSheet
    {
        public string Name { get; set; }
        public decimal RegularLabourRate { get; set; }
        public decimal OvertimeLabourRate { get; set; }
        public decimal PremiumLabourRate { get; set; }
        public decimal RegularTravelRate { get; set; }
        public decimal OvertimeTravelRate { get; set; }
        public decimal PremiumTravelRate { get; set; }

        // Expense rates associated with this sheet
        public decimal HotelCost { get; set; }
        public decimal PerDiemRate { get; set; }
        public decimal MileageRate { get; set; }
        public decimal RentalCarRate { get; set; }
        public decimal FlightCost { get; set; } // Typical flight cost associated with this rate profile

        public RateSheet()
        {
            // Parameterless constructor required for XML deserialization
            Name = "Unnamed Rate Sheet"; // Default name
        }

        public RateSheet(string name)
        {
            Name = name;
            // Initialize with default values or leave as 0 to be set explicitly
            RegularLabourRate = 0;
            OvertimeLabourRate = 0;
            PremiumLabourRate = 0;
            RegularTravelRate = 0;
            OvertimeTravelRate = 0;
            PremiumTravelRate = 0;
            HotelCost = 0;
            PerDiemRate = 0;
            MileageRate = 0;
            RentalCarRate = 0;
            FlightCost = 0;
        }
    }
}

File: src/models/RateSheet.ts
export interface RateSheet {
  name: string;
  regularLabourRate: number;
  overtimeLabourRate: number;
  premiumLabourRate: number;
  regularTravelRate: number;
  overtimeTravelRate: number;
  premiumTravelRate: number;
  
  // Expense rates associated with this sheet
  hotelCost: number;
  perDiemRate: number;
  mileageRate: number;
  rentalCarRate: number;
  flightCost: number; // Typical flight cost associated with this rate profile
}

export class RateSheetClass implements RateSheet {
  name: string = 'Unnamed Rate Sheet';
  regularLabourRate: number = 0;
  overtimeLabourRate: number = 0;
  premiumLabourRate: number = 0;
  regularTravelRate: number = 0;
  overtimeTravelRate: number = 0;
  premiumTravelRate: number = 0;
  hotelCost: number = 0;
  perDiemRate: number = 0;
  mileageRate: number = 0;
  rentalCarRate: number = 0;
  flightCost: number = 0;

  constructor(name?: string) {
    if (name) {
      this.name = name;
    }
  }
} 